### PR TITLE
Added a program for T-prime check(#3869)

### PR DESCRIPTION
**Pull request by ayushjha952**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

### CODE REVIEW

**Review by rohanbabbar04 — COMMENTED** [review]
1) Define functions don't put the entire code in main. Use main for driver code
2) Add sample output after the code not before
3) Add space and time complexity

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * Input:
+ * 49
+ * Output:
+ * 49 is T-prime;
+*/
+
+/*
+*Time complexity : O(sqrt(n)) 
+*Space complexity : O(1)
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Add a new line at the EOF

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    cout<< "Enter the number: "<<endl;
+    cin>>num;
+
+    int result = Check_T_Prime( num );
+
+    if( result == 1 )
+        cout<<num<<" is T-prime. "<<endl;
+        
+    else
+        cout<<num<<" is not T-prime. "<<endl;
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Format the Code

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+               
+                if(square_root % i == 0) {
+                    perfect_square_prime=1;
+                    break;
+                }
+            }
+            
+            //If it is perfect square and square root of perfect square is prime.
+            if (perfect_square_prime == 0)
+            return 1;
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Indent it

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            //If it is perfect square and square root of perfect square not prime.
+            else
+            return 0;
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Indent it

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/check_T_prime.cpp (hunk 2 above)
Comment (ayushjha952) [reply C-Plus-Plus/cp/check_T_prime.cpp]: Here what I have to change can you tell properly.

File: C-Plus-Plus/cp/check_T_prime.cpp (hunk 4 above)
Comment (ayushjha952) [reply C-Plus-Plus/cp/check_T_prime.cpp]: Can you check once I have made the changes which you asked for.

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+{
+
+    //To deal with large numbers
+    long long int num;
+    cout << "Enter the number: " << endl;
+    cin >> num;
+
+    int result = Check_T_Prime(num);
+
+    //If functions returns 1 it is T-Prime.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Redundant Comment

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    if (result == 1)
+        cout << num << " is T-prime. " << endl;
+    //If function returns 0 it is not T-prime.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Redundant Comment

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        //If it is not perfect square and square root of perfect square not prime.
+        else
+            return 0;
+    }
+}
+
+int main()
+{
+
+    //To deal with large numbers
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Redundant Comment

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+            }
+            //If it is perfect square and square root of perfect square is prime.
+            if (perfect_square_prime == 0)
+                return 1;
+            //If it is perfect square and square root of perfect square not prime.
+            else
+                return 0;
+        }
+
+        //If it is not perfect square and square root of perfect square not prime.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Redundant Comment

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+                Because for T-prime number should have 3 factors(one the number itself)
+                and other 2 from square root as it prime number.  
+                */
+                if (square_root % i == 0)
+                {
+                    perfect_square_prime = 1;
+                    break;
+                }
+            }
+            //If it is perfect square and square root of perfect square is prime.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Redundant Comment

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+            if (perfect_square_prime == 0)
+                return 1;
+            //If it is perfect square and square root of perfect square not prime.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Redundant Comment

File: C-Plus-Plus/cp/check_T_prime.cpp [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+ *
+*/
+
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int Check_T_Prime(int n)
+{
+    //Because 2 and 3 are prime(2 factors) and 1 having only 1 factor.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/check_T_prime.cpp]: Redundant Comment

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Comment by HarshCasper** [comment]
@all-contributors bot please add @ayushjha952 for code

**Pull request merged**